docs/src/converting_zygoterules.md

```diff
@@ -0,0 +1,136 @@
+The `ZygoteRules.@adjoint` macro automagically[^1] inserts a extra `nothing` in the return for the function it generates to represent the derivative of output with respect to the function object.
+ChainRules as a philosophy avoids magic as much as possible, and thus require you to return it explicitly.
+If it is a plain function (like `typeof(sin)`), then the differential will be [`NoTangent`](@ref).
+
+
+[^1]: unless you write it in functor form (i.e. `@adjoint (f::MyType)(args...)=...`), in that case like for `rrule` you need to include it explictly.
+
+## Tangent Type changes
+ChainRules uses tangent types that must represent vector spaces (i.e. tangent spaces).
+They need to have things like `+` defined on them.
+ZygoteRules takes a more adhoc approach to this.
+
+### `nothing` becomes an `AbstractZero`
+ZygoteRules uses nothing to represent some sense of zero, in a primal type agnostic way.
+There are many senses of zero.
+ChainRules represents two of them, as subtypes of [`AbstractZero`](@ref).
+
+[`ZeroTangent`](@ref) for the case that there is no relationship between the primal output and the primal input.
+[`NoTangent`](@ref) for the case that conceptually the tangent space don't exist.
+e.g. what is the Tangent to a String or an index: those can't be perturbed.
+
+See [FAQ on difference between `ZeroTangent` and `NoTangent`](@ref faq_abstract_zero).
+At the end of the day it doesn't matter too much if you get them wrong.
+`NoTangent` and `ZeroTangent` more or less act identically.
+
+### `Tuple`s and `NamedTuple`s become `Tangent{T}`s
+Zygote uses `Tuple`s and `NamedTuple`s to represent the structural tangents for `Tuple`s and `struct`s respectively.
+ChainRules core provides a generic `Tangent{T}`(@ref Tangent) to represent the structural tangent of a primal type `T`.
+It takes positional arguments if representing tangent for a `Tuple`.
+Or keyword argument to represent the tangent for a `struct`.
+When representing a `struct` you only need to list the nonzero fields -- any not given are implicit considered to be [`ZeroTangent`](@ref).
+
+When we say structural tangent we mean tangent types that are based only on the structure of the primal.
+This is in contrast to a natural tangent which captures some knowledge based on what the primal type represents.
+(E.g. for arrays a natural tangent is often the same kind of array).
+For more details see the the [design docs on the many tangent types](@ref manytypes)
+
+
+## Calling back into AD (`ZygoteRules.pullback`)
+Rules that need to call back into the AD system, e.g, for higher order functions like `map(f, xs)`, need to be changed.
+In `ZygoteRules` you can use `ZygoteRules.pullback` or `ZygoteRules._pullback`, which will always result in calling into Zygote.
+Since ChainRules is AD agnostic, you can't do that.
+Instead you use a [`RuleConfig`](@ref) to specify requirements of an AD system e.g `::RuleConfig{>:CanReverseMode}` work for Zygote,
```

Comment:
```suggestion
Instead you use a [`RuleConfig`](@ref) to specify requirements of an AD system e.g `::RuleConfig{>:HasReverseMode}` work for Zygote,
```

Reply:
should we also add `NoForwardsMode`?

Reply:
No, you need both when defining a subtype of `RuleConfig`.
But you only need both if you have to to break ambiguity when defining a rule

Reply:
oh of course, sorry for the noise